### PR TITLE
JAMES-3377 Email/query implement minSize/maxSize FilterCondition

### PR DESCRIPTION
**Pull request by chibenwa**

### CODE REVIEW

**Review by rouazana — COMMENTED** [review]
You are missing several tests. I think of:
- minsize only
- maxsize only
Also your test tests too many things, you should demonstrate separately that:
- minsize is inclusive
- maxsize is exlusive

**Comment by chibenwa** [comment]
> You are missing several tests. I think of:
>
>    minsize only
>    maxsize only
>    Also your test tests too many things, you should demonstrate separately that:
>    minsize is inclusive
>    maxsize is exlusive

I agree with the statement that this test tests too many things.

I agree with splitting minsize testing from maxsize

However I disagree adding separate tests for inclusivity/exclusivity.

File: server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailQuery.scala [hunk 1]
```diff
@@ -19,15 +19,19 @@
...
 
+import org.apache.james.jmap.mail.Email.Size
 import org.apache.james.jmap.model.{AccountId, CanCalculateChange, Keyword, Limit, Position, QueryState, UTCDate}
 import org.apache.james.mailbox.model.{MailboxId, MessageId}
 
 case class FilterCondition(inMailbox: Option[MailboxId],
                            inMailboxOtherThan: Option[Seq[MailboxId]],
                            before: Option[UTCDate],
                            after: Option[UTCDate],
+
```
Comment (Arsnael) [inline server/protocols/jmap-rfc-8621/src/main/scala/org/apache/james/jmap/mail/EmailQuery.scala:30]: No need?

**Review by Arsnael — APPROVED** [review]

**Review by remk — APPROVED** [review]

**Comment by chibenwa** [comment]
Just rebased ;-)

**Comment by chibenwa** [comment]
`RabbitMQEventBusTest$ConcurrentTest.concurrentDispatchKeyShouldDeliverAllEventsToListenersWithSingleEventBus » RetryExhausted`

unrelated

test this please

**Comment by chibenwa** [comment]
Deployment tests

Can be considered :green_apple:

test this please anyway

**Comment by chibenwa** [comment]
Merged

**Pull request closed without merging**